src/main/java/com/uid2/admin/secret/SaltRotation.java

```diff
@@ -5,52 +5,34 @@
 import com.uid2.shared.store.salt.RotatingSaltProvider;
 import io.vertx.core.json.JsonObject;
 
-import java.time.Duration;
-import java.time.Instant;
+import java.time.*;
+import java.time.temporal.ChronoUnit;
 import java.util.*;
 import java.util.stream.IntStream;
 
 import static java.util.stream.Collectors.toList;
 
 public class SaltRotation implements ISaltRotation {
-    private static final String SNAPSHOT_ACTIVATES_IN_SECONDS = "salt_snapshot_activates_in_seconds";
-    private static final String SNAPSHOT_EXPIRES_AFTER_SECONDS = "salt_snapshot_expires_after_seconds";
-
     private final IKeyGenerator keyGenerator;
-    private final Duration snapshotActivatesIn;
-    private final Duration snapshotExpiresAfter;
-
-    public static Duration getSnapshotActivatesIn(JsonObject config) {
-        return Duration.ofSeconds(config.getInteger(SNAPSHOT_ACTIVATES_IN_SECONDS));
-    }
-    public static Duration getSnapshotExpiresAfter(JsonObject config) {
-        return Duration.ofSeconds(config.getInteger(SNAPSHOT_EXPIRES_AFTER_SECONDS));
-    }
+    private static final int DAY_IN_SECONDS = 86400;
 
-    public SaltRotation(JsonObject config, IKeyGenerator keyGenerator) {
+    public SaltRotation(IKeyGenerator keyGenerator) {
         this.keyGenerator = keyGenerator;
-
-        snapshotActivatesIn = getSnapshotActivatesIn(config);
-        snapshotExpiresAfter = getSnapshotExpiresAfter(config);
-
-        if (snapshotActivatesIn.compareTo(snapshotExpiresAfter) >= 0) {
-            throw new IllegalStateException(SNAPSHOT_EXPIRES_AFTER_SECONDS + " must be greater than " + SNAPSHOT_ACTIVATES_IN_SECONDS);
-        }
     }
 
     @Override
     public Result rotateSalts(RotatingSaltProvider.SaltSnapshot lastSnapshot,
```

Comment:
I think nextEffective can be passed in, it will allow us to simulate this function more easily over extended time periods and will make it more testable

Reply:
Yeah, let's do that as a separate PR but can be part of this jira. Here's my definition of it in the speclet: https://thetradedesk.atlassian.net/wiki/spaces/UID2/pages/444895602/UID2-4213+Salt+Rotation+Changes+to+Support+V3+Identity+Map#UID2-4213SaltRotationChangestoSupportV3IdentityMap-target_datequeryparam(new,optional)

Reply:
TargetData isn't now though, it's always a UTC midnight. By default it's the next UTC midnight.